### PR TITLE
RCT-101: React Basics/Components, Props, and States

### PR DESCRIPTION
**Pull request by rchie01**
# Related Tickets

https://codeandtheory.atlassian.net/browse/RCT-129

# Description

- Creates a scoreboard using Components, Props, States, and Events in React

# Validation

- Clone the repository using SSH or HTTPS
- Open the RCT-101 folder
- Go Live using the **Live Server** extension in VSCode
- Interact with the scoreboard by removing players (hover over on the left side of the player's name), and adding and deducting player points by using the buttons.

The scoreboard, along with its basic features as guided by the training module should be observed. 

# Risks
- N/A

### CODE REVIEW

File: RCT-101/app.css [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    background: #EF5350;
+  }
+
+  .counter-score {
+    flex-grow: 1;
+    text-align: center;
+    font-family: monospace;
+    font-size: 2em;
+    line-height: 2em;
+  }
```
Comment (lycoangelo) [inline RCT-101/app.css]: Indentation here is not consistent with the others. Also, no new line at EOF.

File: RCT-101/app.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
+const Header = (props) =>{
+    console.log(props);
```
Comment (lycoangelo) [inline RCT-101/app.js]: Stray log.

File: RCT-101/app.js
Comment (lycoangelo) [inline RCT-101/app.js:1]: Please convert tabs to two spaces.

File: RCT-101/app.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+    deductPoint = () => {
+        if(this.state.playerScore > 0 ){
+            this.setState( (prevState) => ({
+                playerScore: prevState.playerScore - 1
+            }));
+        }
+    };
+
+    render(){
```
Comment (lycoangelo) [inline RCT-101/app.js]: Missing space after the parenthesis.

File: RCT-101/app.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        return (
+            <div className="counter">
+                <button className="counter-action decrement" onClick = {() => this.deductPoint()}>-</button>
+                <span className="counter-score">{this.state.playerScore}</span>
+                <button className="counter-action increment" onClick = {() => this.appPoint()}>+</button>
```
Comment (lycoangelo) [inline RCT-101/app.js]: `onClick={this.deductPoint}` should simplify the event callback assignment here.

File: RCT-101/app.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+            </div>
+        );
+    };
+};
+
+class App extends React.Component {
+
+    constructor(){
```
Comment (lycoangelo) [inline RCT-101/app.js]: Missing space after the parenthesis.

File: RCT-101/app.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+        };
+    };
+
+    handleRemovePlayer = (id) => {
+        this.setState( (prevState) => ({
+            players: prevState.players.filter(player => player.id !== id)
+        }));
+    };
+    
+    render(){
```
Comment (lycoangelo) [inline RCT-101/app.js]: Missing space after the parenthesis.

File: RCT-101/app.js [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+        return (
+            <div className="scoreboard">
+                <Header 
+                    header="Scoreboard" 
+                    activePlayers = {this.state.players.length}
+                />
+    
+                {/* Players List */}
+                { this.state.players.map( player => {
+                    return(
```
Comment (lycoangelo) [inline RCT-101/app.js]: Missing space after the `return` keyword.

File: RCT-101/app.js [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+                        <Player 
+                        name={player.playerName}
+                        id={player.id}
+                        key={player.id.toString()}
+                        removePlayer={this.handleRemovePlayer}
+                       
+                        />
```
Comment (lycoangelo) [inline RCT-101/app.js]: Please fix indentations and unnecessary empty lines.

File: RCT-101/app.js [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+    return (
+        <div className="player">
+            <span className="player-name">
+            <button className="remove-player" onClick={() => props.removePlayer(props.id)}>✖</button>
+                { props.name }
+            </span>
+            <Counter />
+        </div>
+    );
+};
```
Comment (lycoangelo) [inline RCT-101/app.js]: Could be an implicit return.

**Review by lycoangelo — CHANGES_REQUESTED** [review]

**Comment by rchie01** [comment]
The aforementioned changes requested above were addressed in the most recent commit. @lycoangelo

**Pull request closed without merging**